### DISCUSSION TITLE
Faster way to get community per each node id?

### DISCUSSION THREAD

**Discussion by Ishitori** [networkit/networkit#1051] [Q&A] [answered]
I am using parallelLeiden to find communities of my undirected, weighted graph (200M nodes, 600M edges).

I use 96 core machine with default settings, and running of ParallelLeiden takes a bit more than 1 minute. That's awesome! But then I need to collect results to a np.array of size number of nodes where np.array[i] returns the community_id for node with that id. Currently, I do it by iterating over `getSubsetIds`:

```
  partitions = np.empty(config.num_nodes, dtype=np.int32)
  subsets = result.getSubsetIds()
  
  log_message(f'Start extracting results. Total communities found: {len(subsets)}')
  for idx, partition_id in enumerate(subsets):
      if idx % 1000 == 0:
          log_message(f'Process partition: {idx}/{len(subsets)}')
          
      for node_id in result.getMembers(partition_id):
          partitions[node_id] = partition_id

```
The thing is that this is slow... I was wondering, is there an API to get an array of partitions faster?

**Comment by fabratu** [comment]
If that is suitable for your use-case - one way to get the information is to retrieve the partition object directly:

```
import networkit as nk
G = nk.generators.BarabasiAlbertGenerator(4,100).generate()
print(G)

partition = nk.community.ParallelLeiden(G).run().getPartition()

print(partition)
```

With this, you can use all functions from that structure object[1].

[1] https://networkit.github.io/dev-docs/python_api/structures.html#networkit.structures.Partition

**Reply by Ishitori** [reply]
Thanks, will try it out

**Comment by Ishitori** [comment, marked as answer]
Turns out the fastest way is to get it using `partition.getVector()` - this returns a list of size Nodes where the value is partition_id
